tiny_dnn/layers/convolutional_layer.h

```diff
@@ -369,6 +384,11 @@ class convolutional_layer : public feedforward_layer<Activation> {
         params_.w_stride = w_stride;
         params_.h_stride = h_stride;
         params_.tbl      = tbl;
+
+        // init padding buffer
+        if (params_.pad_type == padding::same) {
+			cws_.prev_delta_padded_.resize(1, vec_t(params_.in_padded.size(), float_t(0)));
```

Comment:
@edgarriba 
We need NxKx(W-padded)x(H-padded) storage for ```prev_delta_padded_``` but this line reserve 1xKx(W-padded)x(H-padded). We can override ```set_sample_count``` and reserve appropriate buffer there. see example in [max pooling](https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/layers/max_pooling_layer.h#L162_L165).